### PR TITLE
task/79: implemented individual fhir parsers

### PR DESCRIPTION
**Pull request by ghost**
# FHIR IParser modification 

- added individual parsers for Hapi Fhir implementation based on IParser docs (it is recommended to create a new parser for each instance)

### CODE REVIEW

File: app/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiFhirImplementation.java [hunk 1]
```diff
@@ -32,11 +31,13 @@ public <T extends IBase> Optional<T> fhirPathEvaluateFirst(
     @Override
     public <T extends IBaseResource> T parseResource(
             final String fhirResource, final Class<T> clazz) {
-        return PARSER.parseResource(clazz, fhirResource);
+        IParser resourceParser = CONTEXT.newJsonParser();
```
Comment (tjohnson7021) [inline app/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiFhirImplementation.java:34]: I wasn't sure if the individual parsers needed individual names...?

File: app/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiFhirImplementation.java [hunk 2]
```diff
@@ -32,11 +31,13 @@ public <T extends IBase> Optional<T> fhirPathEvaluateFirst(
...
+        return resourceParser.parseResource(clazz, fhirResource);
     }
 
     @Override
     public String encodeResourceToJson(Object resource) {
-        return PARSER.encodeResourceToString((IBaseResource) resource);
+        IParser encodeResourceParser = CONTEXT.newJsonParser();
```
Comment (tjohnson7021) [inline app/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiFhirImplementation.java:40]: I wasn't sure if the individual parsers needed individual names...?

File: app/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiFhirImplementation.java (hunk 1 above)
Comment (halprin) [reply app/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiFhirImplementation.java:34]: I think it's fine.

File: app/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiFhirImplementation.java (hunk 2 above)
Comment (halprin) [reply app/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiFhirImplementation.java:40]: I think it's fine.

**Review by halprin — APPROVED** [review]

**Pull request merged**